### PR TITLE
Add version to feedback emails

### PR DESCRIPTION
**Pull request by yuliadub**
Also figured we can add to the log ones as well? but can remove if you do not think that is necessary. Tried to consolidate version into a separate file since we use it a few places. Still learning how this is all set up, let me know what you think!

what the email looks like (version is weird and no git revision since running locally in an emulator): 
image

### CODE REVIEW

**Comment by yuliadub** [comment]
aha need to make prettier happy! forgot about that

File: components/screens/menu/Version.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as Application from 'expo-application';
+import * as Updates from 'expo-updates';
+import {getUpdateTimeAsVersionString} from 'hooks/useEASUpdateStatus';
+import {Platform} from 'react-native';
+
+export const getVersionInfoFull = (mixpanelUserId: string, updateGroupId: string): string => {
```
Comment (yuliadub) [inline components/screens/menu/Version.tsx:6]: could either mixpanel or update group Id ever be null and do we care if they are?

File: components/screens/MenuScreen.tsx [hunk 2]
```diff
@@ -93,8 +96,22 @@ export const MenuScreen = (queryCache: QueryCache, avalancheCenterId: AvalancheC
   const {data} = useAvalancheCenterMetadata(avalancheCenterId);
   const menuItems = settingsMenuItems[avalancheCenterId];
 
+  const {
+    preferences: {mixpanelUserId},
+  } = usePreferences();
+  const [updateGroupId] = useState(getUpdateGroupId());
```
Comment (stevekuznetsov) [inline components/screens/MenuScreen.tsx:102]: We're not holding onto this state or mutating it in the future, so I don't think you need the `useState()` here:

```ts
const updateGroupId = getUpdateGroupId();
```

File: components/screens/menu/Version.tsx (hunk 1 above)
Comment (stevekuznetsov) [reply components/screens/menu/Version.tsx:6]: I think at least one of them can be, which is why you're getting the error you're getting above:

>  Argument of type 'string | undefined' is not assignable to parameter of type 'string'.

File: components/screens/menu/Version.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import * as Application from 'expo-application';
```
Comment (stevekuznetsov) [inline components/screens/menu/Version.tsx:1]: nit: this file doesn't contain any JSX, so we usually just use a `.ts` extension for that.

File: components/screens/MenuScreen.tsx (hunk 2 above)
Comment (yuliadub) [reply components/screens/MenuScreen.tsx:102]: total newb question, it seems like it wouldn't be needed here either then? https://github.com/NWACus/avy/pull/647/files#diff-93271258411bbda48ad01e2d70e98d5079555570679f1f32dad049f300874dd4L24

File: components/screens/menu/Version.tsx (hunk 1 above)
Comment (yuliadub) [reply components/screens/menu/Version.tsx:6]: so whats the best way to handle this? would we just pass an empty string to the version function if it is null for some reason or have the version function take untyped variable and handle it possibly being null in there? I played around with this a bit, but couldn't find a solution that looked right to me 🤷

File: components/screens/menu/AboutScreen.tsx [hunk 4]
```diff
@@ -25,13 +26,7 @@ export const AboutScreen = (_: NativeStackScreenProps<MenuStackParamList, 'about
   const openUrl = useCallback(({data}: {data: string}) => void WebBrowser.openBrowserAsync(data), []);
   const copyVersionInfoToClipboard = useCallback(() => {
     void (async () => {
-      await Clipboard.setStringAsync(
-        `Avy version ${Application.nativeApplicationVersion || 'n/a'} (${Application.nativeBuildVersion || 'n/a'})
-Git revision ${process.env.EXPO_PUBLIC_GIT_REVISION || 'n/a'}
-Update group ID ${updateGroupId} (channel: ${Updates.channel || 'development'})
-Update ID ${Updates.updateId || 'n/a'} (platform: ${Platform.OS})
-User ID ${mixpanelUserId || 'n/a'}`,
```
Comment (stevekuznetsov) [inline components/screens/menu/AboutScreen.tsx:33]: @yuliadub seems like we could make `getVersionInfoFull()` accept the MixPanel ID as `string | undefined` and inside of the function, do this `mixpanelUserId || 'n/a'` transform to handle the case where it's missing.

File: components/screens/menu/Version.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import {getUpdateTimeAsVersionString} from 'hooks/useEASUpdateStatus';
+import {Platform} from 'react-native';
+
+export const getVersionInfoFull = (mixpanelUserId: string, updateGroupId: string): string => {
+  const versionTestFull = `Avy version ${Application.nativeApplicationVersion || 'n/a'} (${Application.nativeBuildVersion || 'n/a'})
+Build date ${getUpdateTimeAsVersionString()}
+Git revision ${process.env.EXPO_PUBLIC_GIT_REVISION || 'n/a'}
+Update group ID ${updateGroupId || 'n/a'} (channel: ${Updates.channel || 'development'})
+Update ID ${Updates.updateId || 'n/a'} (platform: ${Platform.OS})
+User ID ${mixpanelUserId || 'n/a'}`;
```
Comment (stevekuznetsov) [inline components/screens/menu/Version.ts:12]: Ah, looks like you are already doing the `|| 'n/a'` - so you should be able to just update the function signature.

File: components/screens/menu/Version.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as Application from 'expo-application';
+import * as Updates from 'expo-updates';
+import {getUpdateTimeAsVersionString} from 'hooks/useEASUpdateStatus';
```
Comment (stevekuznetsov) [inline components/screens/menu/Version.ts:3]: You'll need to rebase - I changed this up recently, sorry!

Comment (yuliadub) [reply components/screens/menu/Version.ts]: whoops, sorry I missed that

**Comment by yuliadub** [comment]
@stevekuznetsov I think this is good to go except for [this.](https://github.com/NWACus/avy/pull/647#discussion_r1436119895)  I kept the useState in here for now. Update groupd ID doesnt show up for me to test on the emulator but if I removed the used state weird things happen, so I am leaning to keep it in there for now.

**Comment by stevekuznetsov** [comment]
@yuliadub sweet! Can you squash you commits down and we'll merge in?

**Pull request merged**